frontend/plugins/resources/src/components/DetailPage/DetailPage.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import {Outlet, useParams} from "react-router-dom";
+import DetailPageHeader from "./DetailPageHeader";
+import ResourcesHeader from "../ResourcesHeader";
+import {Content, Page, Table} from "@backstage/core";
+import {Card, CardContent, LinearProgress} from "@material-ui/core";
+
+import Grid from '@material-ui/core/Grid';
+import VersionDistribution from "./VersionDistribution";
+import VersionList from "./VersionList";
+import {DummyResourceResponse} from "zoo-api";
+import Alert from "@material-ui/lab/Alert";
+
+
+const Detailpage = () => {
+  const params = useParams();
+  const resourceId = params.id;
+
+  const generateChartData = (resources) => {
+    const chartdata = []
+    for (const resource of resources) {
+      chartdata.push({
+        name: resource.version,
+        value: resource.versionList.edges.length
+      })
+    }
+    return chartdata
+  }
+  const getChartColors = (chartData) => {
+    function randomHSL(){
+      return `hsla(${~~(360 * Math.random())},70%,70%,0.8)`
+    }
+
+    const colors = []
+    for (let i = 0; i < chartData.length; i++) {
+      colors.push(randomHSL())
+    }
+
+    return colors
```

Comment:
```js
return Array.from({length: chartData.length}, () => randoHSL())
```
Ref: [MDN: Array.from(), using arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#using_arrow_functions_and_array.from)

Reply:
This looks nicer, tnx.